### ISSUE TITLE
OpenCV resize fails with “error: (-215) ssize.area() > 0 in function resize”

### ISSUE THREAD

**Issue by rezazad68** [neuropoly/spinalcordtoolbox#3179]
I would like to straighten the spin generic dataset with the following steps (`sct_deepseg_sc`, `sct_straighten_spinalcord`), however, I am getting the below mentioned error. I have tried with SC versions` 4.0.1` and `5.0.1` but both failed. 

Command:
`sct_deepseg_sc -i sub-amu02_T2w.nii.gz -c t2`

Error : 
```
OpenCV Error: Assertion failed (dsize.area() > 0) in resize, file /myopt/opencv/opencv-3.0.0/modules/imgproc/src/imgwarp.cpp, line 3214
terminate called after throwing an instance of 'cv::Exception'
what():  /myopt/opencv/opencv-3.0.0/modules/imgproc/src/imgwarp.cpp:3214: error: (-215) dsize.area() > 0 in function resize
```

**Comment by joshuacwnewton** [comment]
Hi @rezazad68. Thank you for reporting this error. :slightly_smiling_face: 

Could you please share the full output log (starting from when you ran the `sct_deepseg_sc` command)?

Also, to confirm, the input image is from [`data-multi-subject`](https://github.com/spine-generic/data-multi-subject/blob/master/sub-amu02/anat/sub-amu02_T2w.nii.gz)? Have you encountered this issue with any other images?

**Comment by jcohenadad** [comment]
I cannot reproduce the error on the exact same data so I suspect there is something wrong with the installation of SCT in your system @rezazad68:

<details><summary>Full output</summary>

```
julien-macbook:~/code/spine-generic/data-multi-subject/sub-amu02/anat $ sct_deepseg_sc -i sub-amu02_T2w.nii.gz -c t2

--
Spinal Cord Toolbox (git-jca/centerline-viewer-572bf61299c3c8ed45d04c820d2290fd1fd040d6)

sct_deepseg_sc -i sub-amu02_T2w.nii.gz -c t2
--

Config deepseg_sc:
  Centerline algorithm: svm
  Brain in image: True
  Kernel dimension: 2d
  Contrast: t2
  Threshold: 0.7
Creating temporary folder (/var/folders/s8/4qnm5q1n261ch35b5kkclsb00000gn/T/sct-20210116205623.796523-rp6abi_u)
Reorient the image to RPI, if necessary...
Finding the spinal cord centerline...
Creating temporary folder (/var/folders/s8/4qnm5q1n261ch35b5kkclsb00000gn/T/sct-20210116205633.207223-ctaed5jf)
To avoid intensity overflow due to convertion to +uint16+, intensity will be rescaled to the maximum quantization scale
Remove temporary files...
rm -rf /var/folders/s8/4qnm5q1n261ch35b5kkclsb00000gn/T/sct-20210116205633.207223-ctaed5jf
Cropping the image around the spinal cord...
Normalizing the intensity...
Segmenting the spinal cord using deep learning on 2D patches...
Reassembling the image...
Resampling the segmentation to the native image resolution using linear interpolation...
Binarizing the resampled segmentation...
Removing small objects above slice #231
Removing small objects above slice #231
Compute shape analysis: 100%|##############| 232/232 [00:00<00:00, 321.73iter/s]
Remove temporary files...
rm -rf /var/folders/s8/4qnm5q1n261ch35b5kkclsb00000gn/T/sct-20210116205623.796523-rp6abi_u

Done! To view results, type:
fsleyes /Users/julien/code/spine-generic/data-multi-subject/sub-amu02/anat/sub-amu02_T2w.nii.gz -cm greyscale /Users/julien/code/spine-generic/data-multi-subject/sub-amu02/anat/sub-amu02_T2w_seg.nii.gz -cm red -a 70.0 &

```

</details>

In additional to @joshuacwnewton suggestions, output of these commands could help us:
~~~
sct_check_dependencies
sct_testing
~~~

**Comment by rezazad68** [comment]
Thanks for the advice. Just hecked the dependencies by sct_check_dependencies command; reinstalled and fixed the issue, now works [fine](https://prnt.sc/wybc7m). It was because of the installation. 
Thanks

**Comment by joshuacwnewton** [comment]
Reopening this issue as a user on the forum encountered this error message as well.

That forum thread can be summarized as follows:

* 4.2.2 has an unrelated bug where `sct_download_data` can't download the PAM50 template.
* We [instructed the user](https://forum.spinalcordmri.org/t/sct-version-4-2-2-download-problems/607/2) to side-step this bug by manually downloading the template.
* Then the user ran into this OpenCV error message, see this log: [sct_checkdependencies422.txt](https://github.com/neuropoly/spinalcordtoolbox/files/5831929/sct_checkdependencies422.txt)


It seems to only crop up with broken installations, but it would be nice to know the exact cause.

**Comment by joshuacwnewton** [comment]
Reopening this issue as a user on the forum encountered this error message as well.

That forum thread can be summarized as follows:

* 4.2.2 has an unrelated bug where `sct_download_data` can't download the PAM50 template.
* We [instructed the user](https://forum.spinalcordmri.org/t/sct-version-4-2-2-download-problems/607/2) to side-step this bug by manually downloading the template.
* Then the user ran into this OpenCV error message, see this log: [sct_checkdependencies422.txt](https://github.com/neuropoly/spinalcordtoolbox/files/5831929/sct_checkdependencies422.txt)

It seems to only crop up with broken installations, but it would be nice to know the exact cause.

**Comment by joshuacwnewton** [comment]
A [StackOverflow thread](https://stackoverflow.com/q/31996367) about this error messages gives all kinds of possible causes, so it may be tricky to pinpoint.

**Comment by joshuacwnewton** [comment]
From the user's log on the forum, the error occurs in the following places: 

```python
   # Traceback for `test_sct_detect_pmj`
   File "/data/u_mkaptan_software/SpinalCord_version422/spinalcordtoolbox-4.2.2/scripts/sct_detect_pmj.py", line 204, in detect
   sct.run(cmd_pmj, verbose=0, is_sct_binary=True)

   # Traceback for `test_sct_get_centerline`    
   File "/data/u_mkaptan_software/SpinalCord_version422/spinalcordtoolbox-4.2.2/spinalcordtoolbox/centerline/optic.py", line 119, in detect_centerline
   sct.run(cmd_optic, is_sct_binary=True, verbose=0)
```

So, it seems like it occurs within our `isct_spine_detect` binary, at the very least.

**Comment by joshuacwnewton** [comment]
Ahh, I think I might know what is happening.

For the forum user, the installation fails at the command `sct_download_data -d PAM50`. This means that any downloads that come after don't happen:

https://github.com/neuropoly/spinalcordtoolbox/blob/7efbbb2db90afdb577f836958413bb70fd29c00e/install_sct#L659

Now, @jcohenadad asked the user to run `sct_check_dependencies`, which you _think_ would verify that all these models are present. But it doesn't!

https://github.com/neuropoly/spinalcordtoolbox/blob/7efbbb2db90afdb577f836958413bb70fd29c00e/spinalcordtoolbox/scripts/sct_check_dependencies.py#L256-L261

In summary:
* The user has the PAM50 template, but is missing the required models.
* I believe this leads to a failures in `isct_spine_detect`.
*  `sct_check_dependencies` isn't thorough enough to catch missing models.

**Comment by joshuacwnewton** [comment]
I have verified that https://github.com/neuropoly/spinalcordtoolbox/issues/3179#issuecomment-762426802 is true, both locally and via the user [here](https://forum.spinalcordmri.org/t/sct-version-4-2-2-download-problems/607/13?u=joshuacwnewton). 

Since this issue is easily fixed by making sure the necessary models are present in the `data/` folder, I'm closing. :slightly_smiling_face:
